src/SshKeyScan.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Spatie\Ssh;
+
+use InvalidArgumentException;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Process;
+
+class SshKeyScan
+{
+    private string $sshPort = '22';
+
+    private string $keyType = 'rsa';
+    /**
+     * @var string
+     */
+    private $path;
+    /**
+     * @var string
+     */
+    private $result;
+    /**
+     * @var resource
+     */
+    private $tmpFile;
+    /**
+     * @var string
+     */
+    private $tmpFilePath;
+
+    /**
+     * SshKeyScan constructor.
+     * @param string $hostAddress
+     * @param string $sshPort
+     * @param string $keyType
+     * @throws InvalidArgumentException
+     */
+    public function __construct(string $hostAddress, ?string $sshPort = '22', ?string $keyType = 'rsa')
+    {
+        if (! $this->systemHasCapability()) {
+            throw new InvalidArgumentException('Could not execute ssh-keyscan on host filesystem');
+        }
+
+        $port = [];
```

Comment:
Move the building up of the options to a separate function, much like what you see in the `Ssh` class.